app/src/main/java/org/simple/clinic/instantsearch/InstantSearchScreen.kt

```diff
@@ -125,64 +126,44 @@ class InstantSearchScreen(context: Context, attrs: AttributeSet) : ConstraintLay
       )
   )
 
-  private val events by unsafeLazy {
-    Observable.mergeArray(
-        allPatientsItemClicks(),
-        searchItemClicks(),
-        searchQueryChanges(),
-        registerNewPatientClicks(),
-        blankBpPassportResults()
-    )
-  }
+  override fun defaultModel() = InstantSearchModel.create(screenKey.additionalIdentifier)
 
-  private val delegate by unsafeLazy {
-    val uiRenderer = InstantSearchUiRenderer(this)
+  override fun uiRenderer() = InstantSearchUiRenderer(this)
 
-    MobiusDelegate.forView(
-        events = events.ofType(),
-        defaultModel = InstantSearchModel.create(screenKey.additionalIdentifier),
-        init = InstantSearchInit(),
-        update = InstantSearchUpdate(),
-        effectHandler = effectHandlerFactory.create(this).build(),
-        modelUpdateListener = uiRenderer::render
-    )
-  }
-
-  private val detaches = detaches()
-
-  override fun onAttachedToWindow() {
-    super.onAttachedToWindow()
-    delegate.start()
-  }
+  override fun bindView(layoutInflater: LayoutInflater, container: ViewGroup?) =
+      ScreenInstantSearchBinding.inflate(layoutInflater, container, false)
 
-  override fun onDetachedFromWindow() {
-    delegate.stop()
-    binding = null
+  override fun events() = Observable
+      .mergeArray(
+          allPatientsItemClicks(),
+          searchItemClicks(),
+          searchQueryChanges(),
+          registerNewPatientClicks(),
+          blankBpPassportResults()
+      ).cast<InstantSearchEvent>()
 
-    activity.window.setSoftInputMode(SOFT_INPUT_ADJUST_RESIZE)
+  override fun createUpdate() = InstantSearchUpdate()
 
-    super.onDetachedFromWindow()
-  }
+  override fun createInit() = InstantSearchInit()
 
-  override fun onSaveInstanceState(): Parcelable {
-    return delegate.onSaveInstanceState(super.onSaveInstanceState())
-  }
+  override fun createEffectHandler() = effectHandlerFactory.create(this).build()
 
-  override fun onRestoreInstanceState(state: Parcelable?) {
-    super.onRestoreInstanceState(delegate.onRestoreInstanceState(state))
+  override fun onActivityCreated(savedInstanceState: Bundle?) {
```

Comment:
`onActivityCreated` is being deprecated in the newer version of the fragment library. We can move this to `onViewCreated` instead.

Reply:
We don't have to set the soft input mode here, we can remove this.

Reply:
Removed them from the screen